application/F3DSystemTools.cxx

```diff
@@ -0,0 +1,44 @@
+#include "F3DSystemTools.h"
+
+#include "log.h"
+
+#include <cstring>
+#include <filesystem>
+
+#if defined(_WIN32)
+// clang-format off
+#include <windows.h>
+#include <libloaderapi.h>
+// clang-format on
+#endif
+#ifdef __APPLE__
+#include <mach-o/dyld.h>
+#endif
+
+namespace fs = std::filesystem;
+
+namespace F3DSystemTools
+{
+//----------------------------------------------------------------------------
+fs::path GetApplicationDirectory()
+{
+#if defined(_WIN32)
+  wchar_t wc[1024] = { 0 };
+  GetModuleFileNameW(NULL, wc, 1024);
+  return fs::path(wc);
```

Comment:
maybe something like ?

```
        std::vector<wchar_t> wc(1024);
        if (GetModuleFileNameW(
          GetModuleHandle(nullptr), wc.data(), static_cast<DWORD>(wc.size())))
        {
          return fs::path(wc);
        }
```

Reply:
I haven't changed this code, it's just moved. I can use a `std::array` though if you prefer.

Reply:
I'd just like it to be coherent as it is present in engine.cxx as well. I will take care of it in another PR though